pkg/controller/elasticsearch/driver/upgrade_predicates.go

```diff
@@ -351,7 +351,7 @@ var predicates = [...]Predicate{
 		},
 	},
 	{
-		// Force an upgrade of all the data nodes before upgrading the last master
+		// Force an upgrade of all the master-ineligible nodes before upgrading the last master-eligible
 		name: "do_not_delete_last_master_if_data_nodes_are_not_upgraded",
```

Comment:
Nit: the name of the predicate does not reflect the new behaviour. Non-master nodes could also be ingest only or ML nodes.